### ISSUE TITLE
Slow config reload

### ISSUE THREAD

**Issue by kpetkovic** [hashicorp/consul#4277]
#### Overview of the Issue

We're using consul as service registry with service checks.
We're updating node metadata every 10 seconds and then we issue consul reload.
After running consul agent more than 24 hours, reload time increases from 0.5 second to 13 seconds.

#### Reproduction Steps

1. Create cron job to reload consul configuration every 10 seconds
2. Measure 'consul reload' time after 24 hours

### Consul info for both Client and Server

```
agent:
        check_monitors = 0
        check_ttls = 0
        checks = 1
        services = 1
build:
        prerelease =
        revision = 5174058f
        version = 1.1.0
consul:
        known_servers = 3
        server = false
runtime:
        arch = amd64
        cpu_count = 8
        goroutines = 40
        max_procs = 8
        os = linux
        version = go1.10.1
serf_lan:
        coordinate_resets = 0
        encrypted = true
        event_queue = 0
        event_time = 21
        failed = 0
        health_score = 0
        intent_queue = 0
        left = 0
        member_time = 191
        members = 6
        query_queue = 0
        query_time = 1

```

### Operating system and Environment details

NAME="Ubuntu"
VERSION="16.04.4 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.4 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial

### Log Fragments
When 'consul reload' is run with strace I saw it hangs on " futex(0x20fcb90, FUTEX_WAIT, 0, {60, 0} <unfinished ...>"
It waits for response of local HTTP server.

**Comment by mkeeler** [comment]
@kpetkovic I think you are running into an issue that was just fixed in the 1.2.0 release regarding appending to some lists during a reload instead of wiping them out and creating fresh (which caused lots of duplicate service registrations, watch registration etc). Could you try that one out and verify if it behaves better?

**Comment by kpetkovic** [comment]
Fixed in version 1.2.0.
Thanks.
